src/v3/src/transformer/oktaVerify/transformOktaVerifyEnrollPoll.ts

```diff
@@ -285,16 +285,24 @@ export const transformOktaVerifyEnrollPoll: IdxStepTransformer = ({
     );
 
     if (deviceMap.setupOVUrl) {
+      let challengeMethod;
+      if (deviceMap.isDesktop) {
+        challengeMethod = CHALLENGE_METHOD.CUSTOM_URI;
+      } else if (deviceMap.platformLC === 'android') {
+        challengeMethod = CHALLENGE_METHOD.APP_LINK;
+      } else {
+        challengeMethod = CHALLENGE_METHOD.UNIVERSAL_LINK;
+      }
+
+      // Setup Okta Verify on same device button
       sameDeviceOVElements.push(
         {
           type: 'OpenOktaVerifyFPButton',
           options: {
             step: '',
             i18nKey: 'oie.enroll.okta_verify.setup.title',
             href: deviceMap.setupOVUrl,
-            challengeMethod: deviceMap.isDesktop
-              ? CHALLENGE_METHOD.CUSTOM_URI
-              : CHALLENGE_METHOD.UNIVERSAL_LINK,
+            challengeMethod,
```

Comment:
How does `UNIVERSAL_LINK` differ from `APP_LINK` and why does iOS not use `APP_LINK` as well?

Reply:
They are basically the same thing functionality wise, just each platform has their own name for it.  IOS calls them `UNIVERSAL_LINK`  and Android called them `APP_LINK`.  When on android (and ios), the link can only be opened with `window.location.assign`.  `Util.redirectWithFormGet` will not work.  We have to specify `APP_LINK` here to avoid falling into this case [here](https://github.com/okta/okta-signin-widget/blob/c53ed6cfcc3248277e4d7d7a3265d47a683d171e/src/v3/src/components/OpenOktaVerifyFPButton/OpenOktaVerifyFPButton.tsx#L68).

I think the names here are more to differentiate the platform context.

Reply:
So in the case you linked, when would we expect to hit `isAndroid() && isUniversalLinkMethod` if Android always calls it `APP_LINK`?

Reply:
in the same device enrollment context, I dont think that should ever happen.  That case was made to match parity with v2 for the OV challenge poll view: https://github.com/okta/okta-signin-widget/blob/2b440a4d0ebddeff3c6b8eb80a0bf3f3de249d1d/src/util/Util.js#L193.  
So there might be a scenario that reaches there in that case, but im not exactly sure when.